### PR TITLE
Complete ticket 11 api articles topic query

### PR DESCRIPTION
**Pull request by OJ423**
Adding topic filter functionality
Tested for 404, wrong query and topics without articles

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -138,6 +138,48 @@ describe("API Articles", () => {
...
   });
+  describe("GET /api/articles?:topic", () => {
+    it('200 should return articles filtered by topic', () => {
+      return request(app)
+      .get('/api/articles?topic=mitch')
+      .expect(200)
+      .then(({body}) => {
+        const articles = body.articles
+        const firstArticle = articles[0]
+        expect(articles.length).toBe(12)
```
Comment (nc-roisin-oreilly) [inline __tests__/app.test.js:149]: A good test would also be to iterate through every article that comes back from the api and make sure that their topic is mitch

File: __tests__/app.test.js [hunk 2]
```diff
@@ -138,6 +138,48 @@ describe("API Articles", () => {
...
+    })
+    it('404 when filtering for a topic that does not exist', () => {
+      return request(app)
+      .get('/api/articles?topic=perception')
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe("This topic does not exist")
+      })
+    })
+    it('200 responds with all articles when attempting to sort with an invalid query', () => {
```
Comment (nc-roisin-oreilly) [inline __tests__/app.test.js:166]: Great edge case

File: __tests__/app.test.js [hunk 3]
```diff
@@ -138,6 +138,48 @@ describe("API Articles", () => {
...
+      return request(app)
+      .get('/api/articles?machines=mitch')
+      .expect(200)
+      .then(({body}) => {
+        expect(body.articles.length).toBe(13)
+      })
+    })
+    it('200 responds with empty array for topics with no articles', () => {
```
Comment (nc-roisin-oreilly) [inline __tests__/app.test.js:174]: Fab

File: controllers/articles.controllers.js [hunk 4]
```diff
@@ -20,8 +20,12 @@ exports.getArticleById = (req, res, next) => {
 };
 
 exports.getAllArticles = (req, res, next) => {
-  selectAllArticles()
-    .then((articles) => {
+  const query = req.query
+  const checkTopics = checkTopicsExists(query.topic)
```
Comment (nc-roisin-oreilly) [inline controllers/articles.controllers.js:24]: Nicely done

**Review by nc-roisin-oreilly — COMMENTED** [review]
Excellent work on this ticket :) PR approved!
